src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt

```diff
@@ -4368,7 +4368,7 @@ Expression CaseWhenExpression() #CaseWhenExpression:
 }
 {
     <K_CASE> { caseCounter++; }
-    [ switchExp=Condition() ]
+    [ switchExp=Expression() ]
```

Comment:
How much does the performance degrade? Where are the tests?

Reply:
There is no performance test for the moment. Do you have a standardized way to perform that king of tests? Do you have existing tests showing the current performances? Also, how to evaluate the tradeoff between a supposed drop in performance and a better support of the SQL language?